docs/design/gang-podgroup.md

```diff
@@ -0,0 +1,128 @@
+# Gang PodGroups
```

Comment:
Prefer to `SubGroups of PodGroup`, or other similar title; as this feature not only used by gang in future.